plugins/polio/js/src/domains/Budget/BudgetProcess/CreateBudgetProcessModal.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { Box } from '@mui/material';
+import { Field, FormikProvider, useFormik } from 'formik';
+
+import {
+    AddButton,
+    ConfirmCancelModal,
+    LoadingSpinner,
+    makeFullModal,
+    useSafeIntl,
+} from 'bluesquare-components';
+
+import MESSAGES from '../messages';
+import { MultiSelect } from '../../../components/Inputs/MultiSelect';
+import { Options } from '../types';
+import { SingleSelect } from '../../../components/Inputs/SingleSelect';
+import { useAvailableRoundsForCreate } from '../hooks/api/useGetBudgetProcessAvailableRounds';
+import { useCreateBudgetProcess } from '../hooks/api/useCreateBudgetProcess';
+import { useCreateBudgetProcessSchema } from './validation';
+
+type Props = {
+    isOpen: boolean;
+    closeDialog: () => void;
+};
+
+const CreateBudgetProcessModal: FunctionComponent<Props> = ({
+    isOpen,
+    closeDialog,
+}) => {
+    const { formatMessage } = useSafeIntl();
+
+    const { data: dropdownsData, isFetching: isFetchingDropdownData } =
+        useAvailableRoundsForCreate();
+
+    const { mutate: confirm } = useCreateBudgetProcess();
+    const schema = useCreateBudgetProcessSchema();
+    const formik = useFormik({
+        initialValues: { country: '', campaign: '', rounds: '' },
+        validationSchema: schema,
+        onSubmit: async values => {
+            confirm(values);
+        },
+    });
+    const allowConfirm =
+        !formik.isSubmitting && formik.isValid && Boolean(formik.values.rounds);
+
+    // Filter "Campaign" values on "Country" change.
+    const [currentCampaignOptions, setCampaignOptions] = useState<Options[]>(
+        [],
+    );
+    useEffect(() => {
+        const campaigns = dropdownsData?.campaigns || [];
+        const filtered = campaigns.filter(
+            i => String(i.country_id) === String(formik.values.country),
+        );
+        formik.setFieldValue('campaign', '');
+        setCampaignOptions(filtered);
+    }, [formik.values.country]);
```

Comment:
unless it breaks the interface, good practice would require to include `dropdownsData?.campaigns` in the dependencies array of `useEffect`
```suggestion
    }, [dropdownsData?.campaigns, formik.values.country]);
```

Reply:
@quang-le I can see the ESLint warning but I don't understand this one.

Why is it a good practice to include `dropdownsData?.campaigns` in the dependencies array of `useEffect`?

What is the benefit of adding it?